src/ui/DynamicFacet/DynamicFacet.ts

```diff
@@ -511,10 +548,20 @@ export class DynamicFacet extends Component {
   }
 
   public createDom() {
+    this.createPadding();
     this.createContent();
     this.updateAppearance();
   }
 
+  private createPadding() {
+    const hasFacetColumnParent = !!$$(this.element).parent('coveo-facet-column');
+    if (!this.options.preservePosition || !hasFacetColumnParent) {
+      return;
```

Comment:
Should log an info message here that explains why the facet padding feature will be deactivated.